lm_eval/api/model.py

```diff
@@ -114,6 +114,19 @@ def generate_until(self, requests) -> List[str]:
         """
         pass
 
+    @abc.abstractmethod
+    def apply_chat_template(self, chat_history: list[dict[str, str]]) -> str:
+        """
+        Defines how to transform few-shot examples provided as chat history into a format that can be used as input to the LM.
+
+        :param chat_history: list[dict[str, str]]
+            A list of dictionaries with keys 'role' and 'content'.
+            Values are strings representing the role name and the content of the message, respectively.
+        :return: str
+            A string representing the chat history in a format that can be used as input to the LM.
+        """
+        pass
```

Comment:
Let's not have this be an abstractmethod (it'd break LMs where people don't implement this method) but instead have it just raise NotImplementedError. 

So no *error* will be raised if some LM subclass is defined without this method, but if they try to run using a chat template, we throw an error. How's this sound?